File: test_browser.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  page.on('console', msg => console.log(`[CONSOLE] ${msg.type()}: ${msg.text()}`));
  page.on('requestfailed', request =>
    console.log(`[NETWORK FAIL] ${request.url()}: ${request.failure().errorText}`)
  );
  page.on('response', response => {
    if (!response.ok()) {
      console.log(`[NETWORK HTTP ERROR] ${response.status()} ${response.url()}`);
    }
  });

  try {
    await page.goto('http://localhost:8125/pixel/?bundle=3bb232f739244202', { waitUntil: 'networkidle' });
    await page.waitForTimeout(5000); // wait for phaser to load things
    await page.screenshot({ path: '/tmp/playwright_screenshot.png' });
    console.log('Screenshot saved to /tmp/playwright_screenshot.png');
  } catch (e) {
    console.error('Error during goto:', e);
  }

  await browser.close();
})();

File: test_browser.py
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.firefox.launch(headless=True)
        page = await browser.new_page()
        
        # Listen for console logs
        page.on("console", lambda msg: print(f"Browser console: {msg.type}: {msg.text}"))
        
        print("Navigating to URL...")
        await page.goto("http://127.0.0.1:8125/pixel/?bundle=3bb232f739244202")
        print("Waiting for network idle...")
        await page.wait_for_load_state("networkidle")
        print("Waiting 5s for Phaser...")
        await asyncio.sleep(5)
        print("Taking screenshot...")
        await page.screenshot(path="/tmp/pixel_map_test.png")
        print("Screenshot saved to /tmp/pixel_map_test.png")
        await browser.close()

if __name__ == "__main__":
    asyncio.run(main())
